File: src/MrHotel/WebApp/src/app/services/inventory-manager/data/product-stock-creation-result.ts
import { Guid } from "@customTypes/guid";

export class ProductStockCreationResult {
  constructor(
    private readonly _stockId: Guid,
    private readonly _productId: Guid,
  ) {}

  get stockId(): Guid {
    return this._stockId;
  }

  get productId(): Guid {
    return this._productId;
  }
}

File: src/MrHotel/ApiService/Inventory/Data/ProductStockCreationResult.cs
namespace MrHotel.ApiService.Inventory.Data;

using MrHotel.Database.Entities.Inventory;

public record ProductStockCreationResult(
    Guid StockId,
    Guid ProductId)
{
    public static ProductStockCreationResult FromStock(ProductStock stock) => new(stock.Id, stock.Product.Id);
}
